### DISCUSSION TITLE
Possible to support special characters in Property Name?

### DISCUSSION THREAD

**Discussion by arden828** [alirezanet/Gridify#218] [General]
By using `-` in property will fail the Lex Parsing. Just want to check, currently it is only support Letter, `_` and `.`

The following code will show `"bad character input: '-', at index 8"` in syntaxTree.Diagnostics

```C#
var syntaxTree = SyntaxTree.Parse("property-name = value", []);
```

Any chance to extend it? or make it customisable? As we are using "-" in db columns and we are building customised QueryBuilder as well. 

Thanks

**Comment by alirezanet** [comment]
Hi @arden828,
This may be a concealed bug, as the field section should potentially support the '-' character. I will create a bug issue to investigate this further.

**Reply by alirezanet** [reply]
Fixed in `v2.15.0`
